src/enumerable.cr

```diff
@@ -2292,7 +2298,12 @@ module Enumerable(T)
     # if the type is a union.
     def self.first
       {% if X.union? %}
-        {{X.union_types.first}}
+        {{
+          raise("Enumerable#sum() and #product() do not support Union types. " +
+                "Instead, use Enumerable#sum(initial) and #product(initial), " +
```

Comment:
Stylistic suggestion:

```suggestion
          raise("`Enumerable#sum` and `#product` do not support Union types. " +
                "Instead, use `Enumerable#sum(initial)` and `#product(initial)`, " +
```

Reply:
I can make this change.

Out of curiosity, assuming the back ticks are Markdown's code syntax, what is the effect of including them in in error messages (which I assume are processed and rendered as plain text)?

Reply:
In my eyes it makes them more standout and a clear sign we're referencing the method/type definitions, as in the other, doc-related places.